staging/src/k8s.io/client-go/rest/client.go

```diff
@@ -244,7 +244,9 @@ func (c *RESTClient) Patch(pt types.PatchType) *Request {
 
 // Get begins a GET request. Short for c.Verb("GET").
 func (c *RESTClient) Get() *Request {
-	return c.Verb("GET")
+	// Default cache-control to "no-store, no-cache" to make sure we
+	// don't store sensitive information.
+	return c.Verb("GET").SetHeader("Cache-control", "no-store, no-cache")
```

Comment:
My understanding is that controlling caching behavior using HTTP request / response headers is an accepted pattern.

The mozilla [https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control](Cache-Control) doc lists *no-store* and *no-cache* as known request headers with the following attributes:

> no-store
> The cache should not store anything about the client request or server response.

> no-cache
> Forces caches to submit the request to the origin server for validation before releasing a cached copy.